com.unity.netcode.gameobjects/Components/NetworkTransform.cs

```diff
@@ -698,53 +733,105 @@ private void ApplyInterpolatedNetworkStateToTransform(NetworkTransformState netw
                 }
             }
 
-            // Scale Apply
-            if (SyncScaleX || SyncScaleY || SyncScaleZ)
+            // Apply the new scale
+            if (networkState.HasScaleX || networkState.HasScaleY || networkState.HasScaleZ)
             {
                 transformToUpdate.localScale = interpolatedScale;
             }
         }
 
-        private void AddInterpolatedState(NetworkTransformState newState, bool reset = false)
+        /// <summary>
+        /// Only non-authoritative instances should invoke this
+        /// </summary>
+        private void AddInterpolatedState(NetworkTransformState newState)
         {
             var sentTime = newState.SentTime;
 
-            if (reset)
+            // When there is a change in interpolation or if teleporting, we reset
+            if ((newState.InLocalSpace != InLocalSpace) || newState.IsTeleportingNextFrame)
             {
+                InLocalSpace = newState.InLocalSpace;
+
+                // we should clear our float interpolators
+                foreach (var interpolator in m_AllFloatInterpolators)
+                {
+                    interpolator.Clear();
```

Comment:
Did you test this with artificial packet drop and latency spikes? If we're clearing the buffered interpolator each time we're receiving state, this effectively makes the interpolator a simple lerper, it removes all buffering.

Reply:
This only happens when authority changes InLocalSpace or we are teleporting.